### PR TITLE
Created a order Manager

### PR DESCRIPTION
**Pull request by HaroldFS**
- current order property
- order event log for FS Event creation
- updateItem Quantity

### CODE REVIEW

File: swift/ios-shoppe-demo/ModelControllers/OrderManger.swift [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+//  Copyright © 2020 FullSTory All rights reserved.
+//
+
+import Foundation
+
+class OrderManager {
+    static var sharedInstance = OrderManager()
+
+    var currentOrder: Order?
+    var itemQuantities = [String: Int]()
```
Comment (van-fs) [inline swift/ios-shoppe-demo/ModelControllers/OrderManger.swift]: Having `itemQuantities` hold information here means that we have data in the OrderManager itself and also in `currentOrder`.  Given the `Order` object is pretty shallow, does it make more sense for it to also hold item quantity data?  The approach I took in the Angular app was to forego an order model and simply create a list of products in the [manager](https://github.com/fullstorydev/angular-shoppe-demo/blob/master/src/app/services/cart.service.ts).  The `quantity` property on the `Product` doubles as inventory and item quantities.

File: swift/ios-shoppe-demo/ModelControllers/OrderManger.swift [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+//  Created on 4/1/20.
+//  Copyright © 2020 FullSTory All rights reserved.
+//
+
+import Foundation
+
+class OrderManager {
+    static var sharedInstance = OrderManager()
+
+    var currentOrder: Order?
```
Comment (van-fs) [inline swift/ios-shoppe-demo/ModelControllers/OrderManger.swift]: Does swift instantiate this automatically?

File: swift/ios-shoppe-demo/ModelControllers/OrderManger.swift [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    var itemQuantities = [String: Int]()
+
+    func addProduct(_ product: Product) {
+        currentOrder?.items.append(product)
```
Comment (van-fs) [inline swift/ios-shoppe-demo/ModelControllers/OrderManger.swift]: This adds a reference to a `Product` from elsewhere in the app, right?  I've seen this lead to unintended side-effects as the object changes elsewhere in the app.  In the Angular app, the object is cloned and then added into the items list.  One other thing to be aware is that `Product` has a `UIImage` member, and that begins to mix the View into the Model.

File: swift/ios-shoppe-demo/ModelControllers/OrderManger.swift (hunk 1 above)
Comment (HaroldFS) [reply swift/ios-shoppe-demo/ModelControllers/OrderManger.swift]: 🧐 Looking into this its a great idea

File: swift/ios-shoppe-demo/ModelControllers/OrderManger.swift (hunk 2 above)
Comment (HaroldFS) [reply swift/ios-shoppe-demo/ModelControllers/OrderManger.swift]: thats a good question, I'm not sure I can answer that directly, but its nil until I actually set it equal to something

**Review by van-fs — APPROVED** [review]
LGTM

**Pull request merged**